src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -154,6 +154,10 @@ public LaunchCompleteCallback(IProjectThreadingService threadingService, DebugLa
 
             public void OnComplete(int hr, uint debugTargetCount, VsDebugTargetProcessInfo[] processInfoArray)
             {
+                if (_targetProfile is IDebugProfileLaunchTargetsProvider4 targetsProvider4)
+                {
+                    _threadingService.ExecuteSynchronously(() => targetsProvider4.OnAfterLaunchAsync(_launchOptions, _activeProfile, processInfoArray));
```

Comment:
Does this code need an else block to avoid calling _targetProfile.OnAfterLaunchAsync(...)?

Reply:
I wonder the same. Whatever the answer, it'd be good to document the behaviour on the interfaces in question.

Reply:
The below case has an else. I suppose both before/after should behave identically.

Reply:
See this is why we have code reviews...

Reply:
Yeah, there is supposed to be an `else`. Fixed.